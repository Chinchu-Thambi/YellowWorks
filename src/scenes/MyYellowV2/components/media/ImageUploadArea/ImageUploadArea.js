/* eslint-disable react/prop-types */
import React from 'react';
import { useDropzone } from 'react-dropzone';
import uploadImageIcon from '../../../../../assets/icons/image-upload-icon.svg';

/**
 * @type {(onFilesSelected: Function, maxSizeInMB?: number, multiple?: boolean, small?: boolean) => JSX.Element}
 *
 * @param maxSizeInMB - in megabytes. Defaults to 2MB.
 */
const ImageUploadArea = ({
  onFilesSelected, maxSizeInMB = 2, multiple = false, small = false,
}) => {
  const { getRootProps, getInputProps } = useDropzone({
    accept: 'image/jpeg, image/png',
    maxSize: maxSizeInMB * 1024 * 1024, // in bytes
    maxFiles: multiple ? 0 : 1, // 0 means no limit on how many files
    onDropAccepted: (selectedFiles) => onFilesSelected(selectedFiles),
  });

  return (
    <div
      {...getRootProps({
        className: `w-full h-11 bg-violet-100 text-violet-500 hover:text-violet-600 rounded-md flex flex-col items-center justify-center space-y-2 border-dashed border-2 border-violet-300 cursor-pointer ${
          small ? 'text-xs sm:text-sm max-w-14' : 'text-xs sm:text-base'
        }`,
      })}
    >
      <input {...getInputProps()} />
      <img src={uploadImageIcon} alt="Upload" />
      <div className="mt-3 mb-1 text-center">
        <strong>Upload {multiple ? 'photos' : 'a photo'}</strong> or drag and drop
      </div>
      <div className="text-violet-400 text-center">PNG, JPG up to {maxSizeInMB}MB</div>
    </div>
  );
};

export default ImageUploadArea;
