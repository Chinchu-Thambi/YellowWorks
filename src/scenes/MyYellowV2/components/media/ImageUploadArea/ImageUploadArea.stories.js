import React from 'react';
import { number, boolean } from '@storybook/addon-knobs/react';
import ImageUploadArea from './ImageUploadArea';

export const Component = () => {
  const [selectedFiles, setSelectedFiles] = React.useState();
  const isLarge = boolean('Regular sized', true);
  const isSingle = boolean('Single file', true);
  const maxSizeInMB = number('Max File Size in MB', 2);

  return (
    <>
      <ImageUploadArea
        onFilesSelected={(files) => setSelectedFiles(files)}
        small={!isLarge}
        multiple={!isSingle}
        maxSizeInMB={maxSizeInMB}
      />
      {selectedFiles && (
        <div className="bg-fern-200 text-fern-600 mt-4 p-3">
          <div>TESTING INFO</div>
          <div>Selected files</div>
          <ol>
            {selectedFiles.map((file) => (
              <li>{file.name}</li>
            ))}
          </ol>
        </div>
      )}
    </>
  );
};

export default {
  title: 'Manage Journey/Organisms/Image Upload Area',
};
