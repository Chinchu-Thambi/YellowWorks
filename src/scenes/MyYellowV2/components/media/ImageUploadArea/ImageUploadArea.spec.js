/* globals describe, it, expect */
import React from 'react';
import { mount } from 'enzyme';
import ImageUploadArea from './ImageUploadArea';

describe('Image upload area', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<ImageUploadArea onFilesSelected={(f) => 1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
