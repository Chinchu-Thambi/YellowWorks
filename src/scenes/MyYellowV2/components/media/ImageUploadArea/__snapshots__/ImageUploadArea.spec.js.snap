// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image upload area matches snapshot 1`] = `
<ImageUploadArea
  onFilesSelected={[Function]}
>
  <div
    className="w-full h-11 bg-violet-100 text-violet-500 hover:text-violet-600 rounded-md flex flex-col items-center justify-center space-y-2 border-dashed border-2 border-violet-300 cursor-pointer text-xs sm:text-base"
    onBlur={[Function]}
    onClick={[Function]}
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <input
      accept="image/jpeg, image/png"
      autoComplete="off"
      multiple={true}
      onChange={[Function]}
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex={-1}
      type="file"
    />
    <img
      alt="Upload"
      src="test-file-stub"
    />
    <div
      className="mt-3 mb-1 text-center"
    >
      <strong>
        Upload 
        a photo
      </strong>
       or drag and drop
    </div>
    <div
      className="text-violet-400 text-center"
    >
      PNG, JPG up to 
      2
      MB
    </div>
  </div>
</ImageUploadArea>
`;
